system/jlib/jthread.cpp

```diff
@@ -2031,6 +2058,31 @@ protected: friend class PipeWriterThread;
             started.signal();
             throw makeOsException(errno);
         }
+        ConstPointerArrayOf<char> envp;
+        if (env.length())
+        {
+            ForEachItemIn(idx, env)
+            {
+                envp.append(env.item(idx));
+            }
+#ifdef _WIN32
+            char **e = _environ;
+#else
+#if defined (__APPLE__)
+            char **e = *_NSGetEnviron();
+#else
+            char **e = environ;
+#endif
+#endif
```

Comment:
can the above code and the block starting at line 38, go into platform.h , and into platform agnostic char **getEnvVars() or similar?